### PR TITLE
[UP-412] Boarding pass template example

### PR DESCRIPTION
**Pull request by 3ace**

### CODE REVIEW

File: templates/boarding-pass/pdf_boarding_pass.go [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+		log.Fatal(err)
+	}
+
+	// Create QR code image.
+	qrCode, err := createQRCode("https://github.com/unidoc/unipdf-examples/tree/master/templates/boarding-pass", 100, 100)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Draw main content teplate.
```
Comment (adrg) [inline templates/boarding-pass/pdf_boarding_pass.go:60]: Minor typo in the comment: `teplate` instead of `template`. I think this is a typo in other examples as well.
Could you fix those as well please?

File: templates/boarding-pass/pdf_boarding_pass.go [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+	buf := bytes.NewBuffer(nil)
+	if _, err = io.Copy(buf, file); err != nil {
+		return nil, err
+	}
+
+	return buf, nil
+}
+
+// createQRCode creates a new QR code image encoding the provided text, having
+// the specified with and height.
```
Comment (adrg) [inline templates/boarding-pass/pdf_boarding_pass.go:114]: Minor typo in the comment: `with` instead of `width`. I think this is a typo in other examples as well.
Could you fix those as well please?

File: templates/boarding-pass/templates/main.tpl [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+<table columns="4" column-widths="0.1 0.4 0.4 0.1">
+    <table-cell indent="0" rowspan="3" background-color="#ffffff">
+        <image src="path('templates/res/logo.png')" fit-mode="fill-width" margin="10 10 0 0"></image>
+    </table-cell>
+
+    {{template "table-cell-paragraph" (extendDict $props "Rowspan" 1 "Margin" "10 0 0 0" "Font" "helvetica-bold" "FontSize" 10 "Text" "UniDoc Air") }}
+    {{template "table-cell-paragraph" (extendDict $props "Rowspan" 3 "Margin" "0 20" "Align" "right" "Font" "helvetica" "FontSize" 10 "LineHeight" 1.5 "Text" (printf "ETK:\n%s\nReg No:%s" .Etk .RegNumber )) }}
+
+    <table-cell indent="0" rowspan="3" background-color="#ffffff">
+        <image src="qr-code-1" fit-mode="fill-width" margin="00"></image>
```
Comment (adrg) [inline templates/boarding-pass/templates/main.tpl:27]: No need to specify the `margin` attribute as it defaults to `0` anyway.

File: templates/boarding-pass/templates/main.tpl [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+</table>
+{{end}}
+
+{{$props := dict "Colspan" 1 "Rowspan" 1 "Align" "left" "VerticalAlign" "top" "BackgroundColor" "#ffffff" 
+    "BorderColor" "#000000" "BorderTopSize" 0 "BorderBottomSize" 0 "Indent" 0 
+    "Margin" "3 0 5 0" "LineHeight" 1 "Font" "helvetica" "FontSize" 10 "TextColor" "#000000"}}
+
+{{template "boarding-pass-form" . }}
+
+<table columns="2" column-widths="0.5 0.5">
```
Comment (adrg) [inline templates/boarding-pass/templates/main.tpl:86]: There's no need to specify the `column-widths` attribute. They will have equal sizes by default.

File: templates/boarding-pass/templates/main.tpl [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+    {{template "table-cell-paragraph" (extendDict $props "Text" "Seat") }}
+
+    {{template "table-cell-paragraph" (extendDict $props "BorderBottomSize" 1 "Margin" "0" "Font" "helvetica-bold" "FontSize" 12 "TextColor" "#ff0000" "Text" .FlightNumber ) }}
+    {{template "table-cell-paragraph" (extendDict $props "TextColor" "#000000" "Text" .Gate ) }}
+    {{template "table-cell-paragraph" (extendDict $props "TextColor" "#000000" "Text" .Class ) }}
+
+    <table-cell border-width-bottom="1" indent="0" background-color="#ffffff">
+        <table columns="2" column-widths="0.1 0.9">
+            <table-cell indent="0" vertical-align="middle">
+                <image src="path('templates/res/seat_icon.png')" fit-mode="fill-width" margin="0"></image>
```
Comment (adrg) [inline templates/boarding-pass/templates/main.tpl:61]: No need to specify the `margin` attribute as it defaults to `0` anyway.

File: templates/boarding-pass/pdf_boarding_pass.go [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+	c := creator.New()
+	c.SetPageMargins(30, 30, 20, 20)
+
+	// Read main content template.
+	mainTpl, err := readTemplate("templates/main.tpl")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Read boarding pass data json.
```
Comment (adrg) [inline templates/boarding-pass/pdf_boarding_pass.go:48]: Could you rephrase this comment please?
```
Read boarding pass JSON data.
```

**Review by adrg — COMMENTED** [review]
Looks good. Just a couple of minor observations.

**Comment by 3ace** [comment]
@adrg updated as your suggestion. Thanks

**Review by adrg — APPROVED** [review]
Looks good. Thank you for addressing the review points.
